lib/puppet_forge/v3/release.rb

```diff
@@ -38,6 +39,33 @@ def download(path)
         end
       end
 
+      # Uploads the tarbarll to the forge
+      #
+      # @param path [Pathname] tarball file path
+      # @param api_key [String] Forge API key
+      # @return resp
+      def self.upload(path, api_key)
+        if api_key.nil? 
+          raise "Please provide the api_key for the forge authentication"
+	end
+        encoded_string = Base64.encode64(File.open(path, "rb").read)
+        data = { file: encoded_string }
+
+	resp = self.conn.post do |req|
+                 req.url '/v3/releases'
+                 req.headers['Content-Type']='application/json'
+                 req.headers['Authorization'] = 'Bearer '+api_key
+		         req.body = data.to_json
+                end
+      return resp
+        rescue Faraday::ClientError => e
+          if e.response && e.response[:status] == 403
+            raise PuppetForge::ReleaseForbidden.from_response(e.response)
+          else
+            raise e
+          end
+      end
```

Comment:
It looks like the indentation of this method has gotten a little messed up. :)

Also I think that on success the method should return a new instance of the `V3::Release` class that represents the newly uploaded module release. What do you think?

Reply:
Fixed indentation - sorry, weird vim plugins :)
Also the method now returns self and the response from the upload.